src/main/pairpotentials.cpp

```diff
@@ -32,48 +32,53 @@ void Dissolve::setPairPotentialsIncludeCoulomb(bool b) { pairPotentialsIncludeCo
 bool Dissolve::pairPotentialsIncludeCoulomb() { return pairPotentialsIncludeCoulomb_; }
 
 // Return index of specified PairPotential
-int Dissolve::indexOf(PairPotential *pp) { return pairPotentials_.indexOf(pp); }
+int Dissolve::indexOf(PairPotential *pp)
+{
+    auto result = std::find_if(pairPotentials_.begin(), pairPotentials_.end(), [pp](auto &p) { return pp == p.get(); });
+
+    return result == pairPotentials_.end() ? -1 : result - pairPotentials_.begin();
+}
 
 // Return number of defined PairPotentials
-int Dissolve::nPairPotentials() const { return pairPotentials_.nItems(); }
+int Dissolve::nPairPotentials() const { return pairPotentials_.size(); }
 
 // Add new pair potential to list
 PairPotential *Dissolve::addPairPotential(std::shared_ptr<AtomType> at1, std::shared_ptr<AtomType> at2)
 {
-    PairPotential *pp = pairPotentials_.add();
+    auto &pp = pairPotentials_.emplace_back(new PairPotential());
```

Comment:
```suggestion
    auto &pp = pairPotentials_.emplace_back(std::make_unique<PairPotential>());
```
Is this the line that was causing the issue you were having?  We do exactly this for `Species`, so not sure why it can't be the same here...

Reply:
The issue I was having was that you can't use
```
auto &pp = pairPotentials_.emplace_back();
```

This gives a `nullptr` in `pp`, while this would work with `shared_ptr`